sky/serve/load_balancer.py

```diff
@@ -223,10 +322,250 @@ async def startup():
+                'stat_prefix': 'ingress_http',
+                'http_filters': [{
+                    'name': 'envoy.filters.http.router',
+                    'typed_config': {
+                        '@type': 'type.googleapis.com/envoy.extensions.' +
+                                 'filters.http.router.v3.Router',
+
+                        # We aren't using dynamic_stats, and Envoy recommends
+                        # disabling them for profiling.
+                        'dynamic_stats': False
+                    }
+                }],
+
+                # We don't use random request ids, and Envoy recommends
+                # disabling for profiling.
+                'generate_request_id': False,
+                'route_config': {
+                    'virtual_hosts': [{
+                        'name': 'local_service',
+                        'domains': ['*'],
+                        'routes': [{
+                            'match': {
+                                'prefix': '/'
+                            },
+                            'route': {
+                                'cluster': 'cluster'
+                            }
+                        }]
+                    }]
+                }
+            }
+        }]
+
+        # Listeners are the entry point to envoy.  This one handles all traffic
+        # received on the specified port using the filters described above.
+        # received on port 8080 and processes it with the above filters.
+        listener = {
+            'name': 'listener',
+            'address': {
+                'socket_address': {
+                    'address': '0.0.0.0',
+                    'port_value': self._load_balancer_port,
+                }
+            },
+            'filter_chains': [{
+                'filters': filters
+            }]
+        }
+
+        # These endpoints are the backend IP addresses that the cluster defined
+        # below can load balance over.
+        lb_endpoints = []
+        for url in replicas:
+            # TODO(ejj) when Envoy is the only load balancer choice, we should
+            # just have the controller send us the IP and port rather than
+            # bundling them together in a url.
+            parsed_url = urlparse(url)
+            lb_endpoints.append({
+                'endpoint': {
+                    'address': {
+                        'socket_address': {
+                            'address': parsed_url.hostname,
+                            'port_value': parsed_url.port,
+                        }
+                    }
+                }
+            })
+
+        # A cluster is usually a group of endpoints that can be load balanced
+        # over.  This cluster contains the replicas of the service, and is
+        # handed traffic by the above listener.
+        cluster = {
+            'name': 'cluster',
+            'connect_timeout': '0.25s',
```

Comment:
AI workloads would easily lasts for 10+ seconds. Will this interrupt such a long session?

Reply:
So this timeout governed how long envoy will wait for the TCP connection to be established.  This needs to fully complete before the initial HTTP request kicking off the AI workload begins.  I don't see any particular reason AI workloads would need something different here, but I'm also not sure why I set it.  I may remove it and just go with the default

Reply:
I see. Then it should be fine. Is there any timeout for the whole session length? In our python LB we need to manually set this to a larger value cuz the default value for the session is sth like 10s, which is not sufficient for some AI workload. It would be great to double check does envoy need this setting as well ;)

https://github.com/skypilot-org/skypilot/blob/3bfc29e7819efd320d20e75392bbbca71ded8c4d/sky/serve/load_balancer.py#L148

Reply:
https://www.envoyproxy.io/docs/envoy/latest/faq/configuration/timeouts

Looks like by default the idle timeout (i.e. how long you can have nothing on the stream) is an hour, and the default timeout for active connections is unlimited.  At least for http.

Timeouts for the other stuff seems reasonable as well